htdocs/core/modules/barcode/mod_barcode_thirdparty_standard.php

```diff
@@ -231,17 +231,22 @@ public function getNextValue($objthirdparty = null, $type = '')
 	 *
 	 *	@param	DoliDB		$db					Database handler
 	 *	@param	string		$code				Code to check/correct
-	 *	@param	Societe		$thirdparty			Object third-party
-	 *  @param  int		  	$thirdparty_type   	0 = customer/prospect , 1 = supplier
+	 *	@param	Societe|Product	$thirdparty	Object third party
```

Comment:
I see no reason to allow "Product" here in documentation. In this file method should be called only for Societe.
Why allowing Products here ?

Reply:
Same anwser as for Societe that is allowed:

We have mod_barcode_product.... and mod_barcode_societe... deriving from the same parent class.  The method signatures need to be the same.

Another class could be created, one for barcode_product and one for barcode_thirdparty, but I am focussing on applying typing hints for what they are rather than adding structural changes to fix the notificitions.

In general it would be better to test the class type (instanceof) rather than relying solely on the type/element text field or filename.